update_list.py

```diff
@@ -0,0 +1,351 @@
+            user_agents=parsed_user_agents,
+            browser=browser,
+            version=version,
+            max_version_lag=max_version_lag,
+            limit=limit,
+        )
+
+        if remember:
+            # remember the user agents for next write
+            self.ua_list.extend(filtered_user_agents)
+
+        return filtered_user_agents
+
+    def send_user_agent_request(
+        self,
+        browser: Browser,
+        version: float,
+        limit: int,
+    ) -> list[str]:
+        """Send an API request to get the specified user agents."""
+
+        if browser == Browser.SAFARI:
+            # Specifying a version of a browser does
+            # not work well for Safari. Therefore for
+            # Safari all the available user agents are
+            # downloaded. Selecting a subset with the
+            # correct version happens downstream.
+            input_query_parameters = {
+                "browser": "safari",
+                "browser_type": "browser",
+                "download": "json",
+            }
+        else:
+            input_query_parameters = {
+                "browser": browser.value,
+                "version": str(version),
+                "limit": str(limit),
+                "download": "json",
+            }
+
+        user_agents = self.send_post_request(
+            self.UPDATE_USERAGENT_URL, input_query_parameters
+        )
+
+        return user_agents
+
+    def parse_user_agents(self, user_agents: list[str]) -> list[dict]:
+        """Parse a list of user agent strings and return them as dictionaries."""
+        parsed_user_agents = []
+
+        for user_agent_string in user_agents:
+            try:
+                # replace extra info between square brackets
+                # (usually ip adress) and slashes
+                # and remove trailing whitespace
+                user_agent_string = re.sub(r"\[.*?\]|\\", "", user_agent_string)
+                user_agent_string = user_agent_string.strip()
+
+                # retrieve all the info we need from the user agent string
+                user_agent = parse(user_agent_string)
+                platform = self._get_platform(user_agent)
+                browser = self._get_browser(user_agent)
+                version = self._get_version(user_agent)
+                os = self._get_os(user_agent)
+                system = " ".join(
+                    [user_agent.browser.family, str(version), user_agent.os.family]
+                )
+                ua_parsed = {
+                    "useragent": user_agent_string,
+                    "percent": 100.0,
+                    "type": platform,
+                    "system": system,
+                    "browser": browser,
+                    "version": version,
+                    "os": os,
+                }
+
+                parsed_user_agents.append(ua_parsed)
+            except Exception as e:
+                print(e)
```

Comment:
These prints seem unnecessary. The exception will bubble up and be printed automatically.

Reply:
unless you catch the exception like here.. It will no longer bubble up..?